crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -340,3 +386,142 @@ fn addr(bytes: impl AsRef<[u8]>) -> Result<SuiAddress, Error> {
         Error::Internal(format!("Error deserializing address: {bytes:?}: {e}"))
     })
 }
+
+pub(crate) async fn deserialize_move_struct(
+    move_object: &NativeMoveObject,
+    resolver: &Resolver<PackageCache>,
+) -> Result<(StructTag, MoveStruct), Error> {
+    let struct_tag = StructTag::from(move_object.type_().clone());
+    let contents = move_object.contents();
+    let move_type_layout = resolver
+        .type_layout(TypeTag::from(struct_tag.clone()))
+        .await
+        .map_err(|e| {
+            Error::Internal(format!(
+                "Error fetching layout for type {}: {e}",
+                struct_tag.to_canonical_string(/* with_prefix */ true)
+            ))
+        })?;
+
+    let MoveTypeLayout::Struct(layout) = move_type_layout else {
+        return Err(Error::Internal("Object is not a move struct".to_string()));
+    };
+
+    let move_struct = MoveStruct::simple_deserialize(contents, &layout).map_err(|e| {
+        Error::Internal(format!(
+            "Error deserializing move struct for type {}: {e}",
+            struct_tag.to_canonical_string(/* with_prefix */ true)
+        ))
+    })?;
+
+    Ok((struct_tag, move_struct))
+}
+
+pub(crate) fn get_rendered_fields(
```

Comment:
nit: could you move these helper functions in to the `display` module, rather than the `object` module?